### DISCUSSION TITLE
How long does "Generating SSL Certificate" normally take?

### DISCUSSION THREAD

**Discussion by mikepsinn** [vercel/vercel#5348] [Help] [answered]
Hi!  Thanks for this great service! 😀

I added a domain about 3 hours ago and it still says "Generating SSL Certificate".  How long does that normally take?

![image](https://user-images.githubusercontent.com/2808553/97612903-fe532880-19e5-11eb-9793-074e1c090bd8.png)

**Comment by Sai-Adarsh** [comment]
> Hi! Thanks for this great service! 😀
> 
> I added a domain about 3 hours ago and it still says "Generating SSL Certificate". How long does that normally take?
> 
> ![image](https://user-images.githubusercontent.com/2808553/97612903-fe532880-19e5-11eb-9793-074e1c090bd8.png)

It worked now? because I'm getting the same.

**Reply by mikepsinn** [reply]
It turns out that you have to use their nameservers in order to use a wildcard domain.

**Reply by pn-experience** [reply]
I know this is an old thread, but we just ran into a similar issue. It was fixed by re-pushing the git repo so that Vercel would redeploy. Once it did that, the SSL was generated and active and it worked like a charm.

Hope this helps anyone else!

**Comment by fu-sen** [comment, marked as answer]
I have seen the domain info on whois because you publish the domain here:

```
Domain Name: quantimo.do
   :
Name Server: kia.ns.cloudflare.com
Name Server: kai.ns.cloudflare.com
```

The name server is managed by Cloudflare. This seems to be the cause.
- Wildcard cert need to authenticate the domain with a name server.
- Let's Encrypt will fail domain authentication if it is going through a Cloudflare proxy.

If you don't need to go through this domain via Cloudflare, move your nameservers to Vercel. (recommended)
https://vercel.com/docs/custom-domains

If not, you need to disable Cloudflare's proxy and browse Vercel.
However, you are trying to issue a wildcard certificate, so you will need to take a different action than this method.
https://vercel.com/support/articles/using-cloudflare-with-vercel

**Comment by Ayx03** [comment]
Disable Cloudflare proxy, then delete and re-add the domain(s).
